course.py

```diff
@@ -249,3 +249,45 @@ def add_duplicate_course():
         db.session.rollback()
 
     return redirect(url_for('course.course_info', course_id=course_id, current_year=year))
+
+
+@course_bp.route('/evaluations/<int:user_id>/<int:current_year>')
+@login_required
+def evaluations(user_id, current_year):
+    courses = db.session.query(Course).filter_by(year=current_year).all()
+
+    return render_template('evaluations.html', courses=courses, current_year=current_year, user_id=user_id)
+
+
+@course_bp.route('/create_evaluations/<int:user_id>/<int:current_year>', methods=['POST'])
+@login_required
+def create_evaluation(user_id, current_year):
+    form = request.form
+    if not form:
+        return make_response("Problem with form request", 500)
+
+    course_id = request.form.get('course_id')
+    tasks = request.form.getlist('tasks[]')
+    other_task = request.form.get('other_task')
+    evaluation_hour = request.form.get('evaluation_hour')
+    workload = request.form.get('workload')
+    comment = request.form.get('comment')
+    second_course = request.form.get('second_course') == 'Yes'
+
+    if not all([course_id, evaluation_hour, workload, comment is not None]):
+        return make_response("Missing required fields", 400)
+
+    if other_task:
+        tasks.append(other_task)
+
+    try:
+        new_evaluation = Evaluation(course_id=course_id, course_year=current_year, user_id=user_id, task=tasks,
+                                    nbr_hours=evaluation_hour, workload=workload, comment=comment,
+                                    second_course=second_course)
+        db.session.add(new_evaluation)
+        db.session.commit()
+    except Exception as e:
+        db.session.rollback()
+        raise e
```

Comment:
This exception is propagated but not managed. Actually the form allows to submit an evaluation twice. However the database does not allow it. If one tries to submit the form twice, the exception is thrown.